enhancements/network/ovn-kubernetes-evpn.md

```diff
@@ -0,0 +1,429 @@
+  topology: Layer3
+  layer3:
+    role: Primary
+    subnets:
+      - cidr: "22.100.0.0/16"
+        hostSubnet: 24
+  transport: EVPN
+  evpnConfiguration:
+    vtep: evpn-vtep
+    ipVRF:
+      vni: 100
+---
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: l2-cudn
+  labels:
+    advertise: evpn
+spec:
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: l2-cudn
+  network:
+    topology: Layer2
+    layer2:
+      role: Primary
+      subnets:
+      - "22.100.0.0/16"
+  transport: EVPN
+  evpnConfiguration:
+    vtep: evpn-vtep
+    macVRF:
+      vni: 111
+    ipVRF:
+      vni: 101
+```
+
+### API Extensions
+
+There are no changes required to OCP specific APIs.
+
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+No special considerations for hosted clusters.
+
+#### Standalone Clusters
+
+No special considerations for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+No special considerations for single-node clusters.
+
+### Implementation Details/Notes/Constraints
+
+As a recap, these are the changes proposed by this enhancement:
+
+* Introduce all relevant updates of OVN-Kubernetes and FRR-k8s APIs.
+* Introduce a CNO feature gate for the EVPN feature that will be required to
+  deploy the specific changes of the APIs to configure EVPN.
+
+### Risks and Mitigations
+
+Development of this feature is subject to the availability of FRR v9+. The
+problem and possible mitigations have been described in the Proposal section of
+this enhancement.
+
+Otherwise, the risk and mitigations for this feature are aligned to the ones
+described in the upstream [enhancement][9].
+
+[9]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/4568650412f85cdcf7671c74d2e9b3bb67d3b366/docs/okeps/okep-5088-evpn.md#risks-known-limitations-and-mitigations
+
+### Drawbacks
+
+The drawbacks of this feature are aligned to the ones described in the upstream
+[enhancement][9].
+
+As previously mentioned, the EVPN feature is only supported when
+`routingViaHost` is set to `true`, also known as local gateway mode. Support for
+this configuration will be introduced in a future enhancement.
+
+Also worth noting that the EVPN feature is not compatible with other
+OVN-Kubernetes features like EgressIP or IPsec. The compatibility matrix is
+described with more detail in the upstream [enhancement][10]. Further
+enhancements will be required to resolve these compatibility issues as needed.
+
+[10]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/okeps/okep-5088-evpn.md#feature-compatibility
+
+Among those limitations is the lack of support to advertise service IPs. MetalLB
+would be the primary candidate to support this functionality and while it
+supports unicast advertisements, it does not currently support the EVPN address
+family. Thus, the advertisement of load balancer IPs on EVPN must be configured
+manually until support is added to MetalLB. 
+
```

Comment:
are there any dependencies on other feature gates or downstream specific feature interoperability outside our immediate team (example gatewayAPI or ingress routes support or Mesh or anything worth calling out here)? In the initial UDN design we had a design set for routes implementation but its not fully implemented, but I guess if that's done then for EVPN specifically either we would need to see if things work out of the box or if we need to do anything special I guess.